### PR TITLE
Support for hydrogen adapter

### PR DESCRIPTION
**Pull request by MidhunSureshR**

### CODE REVIEW

File: trafficlight/tests/hydrogen_actions_demo.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            await alice_one.open_room("Test Room 1")
+            await alice_one.reload()
+            await alice_one.verify_message_in_timeline("Hello world!")
+            await alice_one.logout()
+            await alice_one.login(server)
+            await alice_one.open_room("Test Room 1")
+            await alice_one.verify_last_message_is_utd()
+            await alice_one.clear_idb_storage()
+        finally:
+            await nio_client.close()
```
Comment (michaelkaye) [inline trafficlight/tests/hydrogen_actions_demo.py:42]: If there's only one client using the alice user; we should call it alice not alice_one.

```suggestion
        self._client_under_test([HydrogenWeb()], "alice")
        self._server_under_test(Synapse(), ["server"])

    async def run(
        self,
        alice: MatrixClient,
        server: HomeServer,
    ) -> None:
        nio_client = AsyncClient(
            server.cs_api, f"@{alice.localpart}:{server.server_name}"
        )
        try:
            await nio_client.register(alice.localpart, alice.password)
            await alice.login(server)
            await alice.create_room("Test Room 1")
            await alice.send_message("Hello world!")
            await alice.create_room("Test Room 2")
            await alice.open_room("Test Room 1")
            await alice.reload()
            await alice.verify_message_in_timeline("Hello world!")
            await alice.logout()
            await alice.login(server)
            await alice.open_room("Test Room 1")
            await alice.verify_last_message_is_utd()
            await alice.clear_idb_storage()
        finally:
            await nio_client.close()
```

**Review by michaelkaye — APPROVED** [review]
A few minor suggestions; if you're ok with them as-is then this is good to merge.

**Comment by michaelkaye** [comment]
(also merged the main branch in to reduce the changes to review in this PR - would suggest we squash-merge because there's a lot of commits here)

File: trafficlight/homerunner.py [hunk 2]
```diff
@@ -48,12 +48,12 @@ def __init__(self, homerunner_url: str) -> None:
...
     def _generate_homeserver(self, base_image_uri: str) -> Dict[str, Any]:
         """
-        For now, we don't require any Users or Rooms to be created automatically
+        For now, we only need user for hydrogen adapter!
         """
         self.hsid = self.hsid + 1
         return {
             "Name": "trafficlight" + str(self.hsid),
-            "Users": [],
+            "Users": [{"Localpart": "testuser", "DisplayName": "testuser"}],
```
[the remark this answers is not in the document]
Comment (MidhunSureshR) [reply trafficlight/homerunner.py:56]: I wonder if we even need to create users via homerunner anymore since matrix-nio can be used?

**Pull request merged**